packages/jest-expo/src/preset/createMockConstants.js

```diff
@@ -44,22 +42,20 @@ function _readAppConfiguration() {
   do {
     currentDirectory = nextDirectory;
 
-    const candidatePath = path.join(currentDirectory, 'app.json');
-    json = _safeReadFile(candidatePath, 'utf8');
+    try {
+      config = getConfig(currentDirectory);
+    } catch (e) {
+      if (!e.message.includes('expected package.json path')) {
```

Comment:
This seems really fragile. Someone could change the config package or whatever throws this error and break this code that's in a different repo. Is there some way we could avoid getting this error up front, or at least make it part of the public API (e.g. with an error code) of getConfig?